File: php/features.php
<?php
declare(strict_types=1);

// AdressApiFrance SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AdressApiFranceFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AdressApiFranceBaseFeature();
            case "test":
                return new AdressApiFranceTestFeature();
            default:
                return new AdressApiFranceBaseFeature();
        }
    }
}

File: py/adressapifrance_sdk/features.py
# AdressApiFrance SDK feature factory

from adressapifrance_sdk.feature.base_feature import AdressApiFranceBaseFeature
from adressapifrance_sdk.feature.test_feature import AdressApiFranceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AdressApiFranceBaseFeature(),
        "test": lambda: AdressApiFranceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
